src/UI/Document.elm

```diff
@@ -288,6 +291,14 @@ withSidebarStyle style (Document nav) =
     Document { nav | sidebarStyle = style }
 
 
+{-| `Document.withSidebarStyle` takes a `SidebarStyle` setting the
+appearance/behavior of the sidebar when it is enabled by the `hasMenu` flag.
+-}
+withLegacyTransitionMenuStyle : Document page msg -> Document page msg
```

Comment:
I think changing `SidebarStyle` into 

```elm
type SidebarStyle
    = SidebarPersistent
    | SidebarNonPersistent
    | SidebarLegacyMigration
```

would be a better approach

Reply:
In this way i can't have Legacy and Persistent